src/Catty/ViewController/Help/HelpWebViewController.swift

```diff
@@ -142,111 +162,128 @@ class HelpWebViewController: UIViewController, UIWebViewDelegate, UIScrollViewDe
         }
     }
 
-    func webViewDidFinishLoad(_ webView: UIWebView) {
+    func webView(_ webView: WKWebView, didFinish navigation: WKNavigation!) {
         Util.setNetworkActivityIndicator(false)
-        url = webView.request?.url
+        url = webView.url
         setupToolbarItems()
         errorLoadingURL = false
         doneLoadingURL = true
         loadingView?.hide()
         setProgress(1.0)
 
         UIView.animate(withDuration: 0.25, animations: {
-            self.webView.alpha = 1.0
+            self.webView?.alpha = 1.0
         })
     }
 
-    func webViewDidStartLoad(_ webView: UIWebView) {
+    func webView(_ webView: WKWebView, didStartProvisionalNavigation navigation: WKNavigation!) {
         Util.setNetworkActivityIndicator(true)
         doneLoadingURL = false
         setProgress(0.2)
         setupToolbarItems()
     }
 
-    func webView(_ webView: UIWebView,
-                 shouldStartLoadWith request: URLRequest,
-                 navigationType: UIWebView.NavigationType) -> Bool {
+    func webView(_ webView: WKWebView, decidePolicyFor navigationAction: WKNavigationAction, decisionHandler: @escaping (WKNavigationActionPolicy) -> Void) {
```

Comment:
Please add a test case verifying the complex behavior of this method (i.e. `CattyTests/ViewController/HelpWebViewControllerTests.swift`). Important: do not make real web requests, just test the result of the `decisionHandler` by using test doubles if necessary.

Reply:
I had basically converted this method's previous implementation (i.e. from UIWebView's delegate) to fit with delegate methods of WKWebView. Now that I have also written tests for this method, I think this method needs some improvements and can be made simpler by calling other methods. Currently, the changes that I have made are only that I felt are required. For example, the validity of the URL was only being checked when there's a download URL detected, which I came to know after I wrote the test for it. So I moved it above other condition checks in the method. Please let me know if there's anything that I misunderstood or there's anything that can still be improved.